CMakeLists.txt

```diff
@@ -175,6 +175,11 @@ if(BUILD_CLANG_PLUGIN)
       set(ROCM_VERSION_MINOR ${CMAKE_MATCH_2})
       set(ROCM_VERSION_PATCH ${CMAKE_MATCH_3})
       message(STATUS "AMD clang version: ${ROCM_VERSION_MAJOR}.${ROCM_VERSION_MINOR}.${ROCM_VERSION_PATCH}")
+
+      if(${LLVM_VERSION_MAJOR} EQUAL 13 AND NOT DEFINED ${HIPSYCL_NO_DEVICE_MANGLER})
+        message(SEND_ERROR "AMD clang version not compatible with hipSYCL explicit multipass. "
+                           "To disable explicit multipass set -DHIPSYCL_NO_DEVICE_MANGLER=ON.")
```

Comment:
Can we also mention that this breaks simultaneous compilation for both CUDA and HIP? This is probably when most people are going to come into contact with explicit multipass.